pkg/services/ldap/settings.go

```diff
@@ -135,9 +138,28 @@ func readConfig(configFile string) (*Config, error) {
 		}
 	}
 
+	for _, server := range result.Servers {
+		envValue := evalEnvVar(server.BindPassword)
+		server.BindPassword = envValue
+	}
+
 	return result, nil
 }
 
+func evalEnvVar(value string) string {
+	regex := regexp.MustCompile(`\${(\w+)}`)
+	if regex != nil {
+		return regex.ReplaceAllStringFunc(value, func(envVar string) string {
```

Comment:
I would suggest using FindSubmatch which would allow you to skip the trimming and use the capturing group you have in your regexp directly.

Example:
```go
package main

import (
	"fmt"
	"regexp"
)

func main() {
	pwd := "${bind_password}"

	regex := regexp.MustCompile(`\${(\w+)}`)
	submatch := regex.FindSubmatch([]byte(pwd))

	match := string(submatch[1])
	fmt.Println(match)
}

# Output:
# bind_password
```

Reply:
I presume this code has been copied from the main settings reader. We should change that as well, but no need to do that in this PR.